app/models/instance.rb

```diff
@@ -9,30 +9,24 @@
 #
 
 class Instance < ApplicationRecord
+  include DatabaseViewRecord
+
   self.primary_key = :domain
 
   attr_accessor :failure_days
 
-  has_many :accounts, foreign_key: :domain, primary_key: :domain, inverse_of: false
-
   with_options foreign_key: :domain, primary_key: :domain, inverse_of: false do
     belongs_to :domain_block
     belongs_to :domain_allow
     belongs_to :unavailable_domain # skipcq: RB-RL1031
+
+    has_many :accounts, dependent: nil
   end
 
   scope :searchable, -> { where.not(domain: DomainBlock.select(:domain)) }
   scope :matches_domain, ->(value) { where(arel_table[:domain].matches("%#{value}%")) }
   scope :by_domain_and_subdomains, ->(domain) { where("reverse('.' || domain) LIKE reverse(?)", "%.#{domain}") }
 
-  def self.refresh
-    Scenic.database.refresh_materialized_view(table_name, concurrently: true, cascade: false)
-  end
```

Comment:
This would replace the `concurrently: true` with `concurrently: false`.

Reply:
Ah I did this quickly and did not catch that the three were not the same.

Are they intentionally different for some background reason ... or could I update the base class to be `concurrently: true` and that'd be acceptable change for the other classes?

Reply:
Looks like as long as you are on PG 9.4+ and the view has at least one unique index (these all do) you can use `concurrently: true`.

Updated to do that here.

Reply:
Doesn't this break if the materialized view was never populated in the first place?

Reply:
Like in a scenario where the migrations were not run and PG doesn't know about the view but you try to refresh it? Or a scenario where the view was created but w/out any relevant data found and you try to refresh it? Something else?

Reply:
We have a bunch of 2021 migrations which declared materialized views with `WITH NO DATA`, which if I remember correctly requires you to populate them without `CONCURRENTLY` at least once before being able to do it `CONCURRENTLY`

Reply:
Ah, I see -- the migrations which pull in the account_summaries view has no_data true, as does a migration which updates follow_recommendations to v2 of it's sql definition. I'll experiment with that locally and see if this breaks prior behavior.

Reply:
Checked and you are correct -- in both main and this PR branch, if you run just the migrations (ie, not from schema:load or db:reset or whatever), the views which have no_data in the migration are not populated.

In main, running `AccountSummary.refresh` as an example, will execute correctly. On this branch, for the no_data views, there's an error like: `PG::FeatureNotSupported: ERROR:  CONCURRENTLY cannot be used when the materialized view is not populated` and this is consistent with PG docs on the feature.

It looks like `Scheduler::FollowRecommendationsScheduler` is the place these are called from.

I'm guessing the background motivation here is that we DO NOT want to run the view population process during the migration process (to keep it fast), and we then regularly run the view refresh process via that scheduled task for these views?

Reply:
> I'm guessing the background motivation here is that we DO NOT want to run the view population process during the migration process (to keep it fast), and we then regularly run the view refresh process via that scheduled task for these views?

That's correct

Reply:
Tangent to the issue here ... where would be the best spot to add coverage so that issues like "rebuilding with concurrent true will fail" would have been caught?

I initially figured the FollowRecommendationScheduler must not have any coverage (it calls their refresh methods), but it turns out it does have coverage ... but I realized that in that scenario the schema is being loaded directly which I think is implicitly generating the materialized views when it loads; whereas going through all the migrations would not do that until the refreshes were actually run directly.

Is this something it'd be reasonable to add into the various data integrity/consistency checks? Basically just have it run refresh on all view classes immediately after migrating to confirm they do so w/out error. I think this would have caught my incorrect assumption here.

Reply:
Updated to use the `rescue` approach from the other related PR. Will rebase this when that merges.